gwcs/legacy_wcs.py

```diff
@@ -0,0 +1,82 @@
+# Module to wrap the old WCS in the gWCS framework
+
+from astropy.modeling import FittableModel, Parameter
+
+
+from astropy.wcs import WCS
+
+
+
+def _get_possible_wcs_parameters(wcs):
+    """
+    get possible Parameters
+    :param wcs:
+    :return:
+    """
+    return [item.replace('get_', '') for item in wcs.__class__.__dict__
+            if item.startswith('get_')]
+
+
+def read_fits_wcs(header=None):
+    """
+    Parameters
+    ----------
+        header : astropy.io.fits header object, string, dict-like, or None, optional
+            If *header* is not provided or None, the object will be
+            initialized to default values.
+    """
+    legacy_wcs = WCS(header)
+
+
+    possible_parameters = _get_possible_wcs_parameters(legacy_wcs.wcs)
+
+    param_dict = {}
+    class_dict = {}
+    for param in possible_parameters:
+        param_value = getattr(legacy_wcs.wcs, 'get_{0}'.format(param))()
```

Comment:
Actually, looking at this deeper, I'd say: get_pc and get_cdelt.  The tricky bit is that we'll also need code to dynamically update those values during fitting, but I wouldn't know how to do that.  @nden or @embray might.